draft-ietf-mls-protocol.md

```diff
@@ -954,11 +961,25 @@ operations:
   is processed
 * The `tree_hash` is updated to represent the current tree and
   credentials
-* The `transcript_hash` is updated by a GroupOperation message
-  `operation` in the following way:
+* The `transcript_hash` is updated with the data for an MLSPlaintext
+  message encoding a group operation in two parts:
 
 ~~~~~
-transcript_hash_[n] = Hash(transcript_hash_[n-1] || operation)
+struct {
+  opaque group_id<0..255>;
+  uint32 epoch;
+  uint32 sender;
+  ContentType content_type;
+  GroupOperation operation;
+} MLSPlaintextOpContent;
+
+struct {
+  opaque confirmation<0..255>;
+  opaque signature<0..2^16-1>;
+} MLSPlaintextOpAuthData;
+
+intermediate_hash_[n] = Hash(transcript_hash_[n-1] || MLSPlaintextOpAuthData_[n-1]);
```

Comment:
I'm not sure we have to introduce the notion of an intermediate hash, since it will never be used independently. We could just define the transcript hash as:

```
transcript_hash_[n] = Hash(transcript_hash_[n-1] || MLSPlaintextOpAuthData_[n-1] || MLSPlaintextOpContent_[n]);
```

Reply:
The idea of the intermediate hash is that an implementation could go ahead and update the transcript hash to this state and be done with MLSPlaintext_[n-1], so it wouldn't need to keep that around..  I'll add some text to clarify.

Reply:
That makes sense